client/engine/engine.go

```diff
@@ -184,6 +191,61 @@ func (e *Engine) handleProposal(proposal consensus_channel.Proposal) (EngineEven
 	return e.attemptProgress(obj)
 }
 
+// updateObjective handles updating and cranking the objective and dispatching any side effects
+func (e *Engine) updateObjective(objective protocols.Objective, event protocols.ObjectiveEvent) (EngineEvent, error) {
+	engineEvent := EngineEvent{}
+	if objective.GetStatus() == protocols.Unapproved {
+		e.logger.Printf("Policymaker is %+v", e.policymaker)
+		if e.policymaker.ShouldApprove(objective) {
+			objective = objective.Approve()
+
+			ddfo, ok := objective.(*directdefund.Objective)
+			if ok {
+				// If we just approved a direct defund objective, destroy the consensus channel to prevent it being used (a Channel will now take over governance)
+				e.store.DestroyConsensusChannel(ddfo.C.Id)
+			}
+		} else {
+			objective, sideEffects := objective.Reject()
+			err := e.store.SetObjective(objective)
+			if err != nil {
+				return EngineEvent{}, err
+			}
+
+			engineEvent.CompletedObjectives = append(engineEvent.CompletedObjectives, objective)
```

Comment:
We should just call `Merge`

Reply:
Actually nevermind! `Merge` only makes sense when we're dealing with two `EngineEvent`s